File: v8_liskov_substitution_principle/python/main2.py
class Calculator:

    def calculate(self, n1, n2):
        return n1 + n2

class SuperCalculator(Calculator):

    sum = 10

    def add(self, n1, n2):
        return n1 + n2

    def substract(self, n1, n2):
        return n1 - n2

sc = SuperCalculator()
print(sc.substract(10, 5))

File: v8_liskov_substitution_principle/java/Main2.java
package v8_liskov_substitution_principle.java;

public class Main2 {
    
}

class Calculator {
	public int calculate(int n1, int n2) { return n1 + n2; }
}

class SuperCalculator extends Calculator {
	private int sum = 10;
	public int add(int n1, int n2) { return n1 + n2;}
	public int subtract(int n1, int n2) { return n1 - n2;}
}


